cpp/include/cuml/fil/multi_reduction.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
```

Comment:
@canonizer I can replace this with the single-loop version for simplicity and we can look for that 1% total speedup elsewhere. There are enough caveats to this 1% that it might make more sense to profile the kernel elsewhere and tune other things.

Reply:
actually, the difference is just due to storing the outputs back in shared memory. I fixed it and now it's both simpler and just as fast.